src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -201,14 +207,24 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
               @Optional() @Inject(DOCUMENT) private _document: any,
               // @breaking-change 8.0.0 Make `_viewportRuler` required.
               private _viewportRuler?: ViewportRuler) {
+    this._scrollStrategy = scrollStrategy;
+  }
 
+  ngAfterViewInit() {
     if (typeof window !== 'undefined') {
-      _zone.runOutsideAngular(() => {
+      this._zone.runOutsideAngular(() => {
         window.addEventListener('blur', this._windowBlurHandler);
       });
-    }
 
-    this._scrollStrategy = scrollStrategy;
+      if (supportsShadowDom()) {
```

Comment:
Is this strictly necessary? Is checking for the existence of `getRootNode` here not enough?

Reply:
`getRootNode` isn't strictly related to Shadow DOM. E.g. calling it on a regular element will return the `document`.